Dockerfiles/agent-ot/Dockerfile.byoc

```diff
@@ -0,0 +1,84 @@
+ARG AGENT_VERSION=nightly-ot-beta-main-jmx
+# Use the Ubuntu Slim AMD64 base image
+FROM ubuntu:24.04 AS builder
+
+# Set environment variables
+ARG AGENT_VERSION
+ENV DEBIAN_FRONTEND=noninteractive
+ARG GO_VERSION=1.22.0
```

Comment:
This is already out of date, the version according to the `.go-version` [file](https://github.com/DataDog/datadog-agent/blob/main/.go-version) is `1.22.5`

Which raises a question about the maintainability of this approach as it currently stands. Maybe we should move the go install to after the agent codebase git checkout, and get the go version from this file?

Reply:
update